numba/cuda/simulator/kernel.py

```diff
@@ -176,7 +175,9 @@ def run(self):
             else:
                 msg = '%s %s: %s' % (tid, ctaid, e)
             tb = sys.exc_info()[2]
-            self.exception = (type(e), type(e)(msg), tb)
+            # Using `with_traceback` here would cause it to be mutated by
+            # future raise statements, which may or may not matter.
+            self.exception = (type(e)(msg), tb)
```

Comment:
@gmarkall do you have a view here? I think what's present should be fine.

Reply:
To elaborate on that commentL using
```
self.exception = e.with_traceback(tb)
```
followed by
```
raise self.exception
```
results in self.exception.__traceback__ being modified in-place, making a second `raise self.exception` give a longer traceback.
I don't know if that can ever happen, but since the code before avoided this issue, I figured I should leave the semantics identical.

Reply:
Thanks for expanding. I can't see a need to change this behaviour.